### PR TITLE
[client] Fall back to getent/id for SSH user lookup in static builds

### LINKED ISSUE #4919: Can't ssh onto a user defined in ldap
**Describe the problem**

When trying to ssh onto a server with it's users defined in a remote ldap netbird failed to resolve the username. 
When trying to ssh directly to the node it works as expected. 

nssswitch config and sssd are properly configured and using id command on the server resolve the remote user without any issue.

**To Reproduce**

try netbird ssh on a server with users defined in remote ldap

**Expected behavior**

being able to ssh into the node

**Are you using NetBird Cloud?**

using netbird cloud

**NetBird version**

0.60.5

**Is any other VPN software installed?**

No

**Debug output**

To help us resolve the problem, please attach the following anonymized status output

 netbird status -dA
Peers detail:
 t480.netbird.cloud:
  NetBird IP: 100.99.150.233
  Public key: obe/7CnEUsrmXTmTVA8gbTcZWwzmhYBTEmRIcRtxzgE=
  Status: Connected
  -- detail --
  Connection type: P2P
  ICE candidate (Local/Remote): srflx/prflx
  ICE candidate endpoints (Local/Remote): 198.51.100.0:51820/198.51.100.3:51820
  Relay server address: rels://streamline-es-mad1-0.relay.netbird.io:443
  Last connection update: 2 hours, 14 minutes ago
  Last WireGuard handshake: 1 minute, 23 seconds ago
  Transfer status (received/sent) 703.1 KiB/2.6 MiB
  Quantum resistance: false
  Networks: -
  Latency: 5.445008ms

Events:
  [INFO] SYSTEM (a5a62818-a1c6-45ba-9ba9-ea9fccb5ed24)
    Message: Network map updated
    Time: 5 hours, 6 minutes ago
  [INFO] SYSTEM (629adc52-ee11-4d64-bcb4-124fdb0243ea)
    Message: Network map updated
    Time: 5 hours, 6 minutes ago
  [INFO] SYSTEM (4c45fe1a-c08c-4eff-b894-2e0c5d26bb89)
    Message: Network map updated
    Time: 5 hours, 1 minutes ago
  [INFO] SYSTEM (9d32ffac-d530-4932-b9c9-a265627ce183)
    Message: Network map updated
    Time: 4 hours, 49 minutes ago
  [INFO] SYSTEM (4875611c-8222-4eb7-a74d-f75b391ae7b0)
    Message: Network map updated
    Time: 4 hours, 49 minutes ago
  [INFO] SYSTEM (cb229fa6-fe01-4a1d-8c61-f031ee57f290)
    Message: Network map updated
    Time: 3 hours, 30 minutes ago
  [INFO] SYSTEM (8c69f207-cc53-4cd9-b12c-a4293ade4044)
    Message: Network map updated
    Time: 44 minutes, 57 seconds ago
  [INFO] SYSTEM (701a0c06-46ff-4d99-b567-3c4d973655fe)
    Message: Network map updated
    Time: 44 minutes, 57 seconds ago
  [INFO] SYSTEM (871a4a3f-7239-4ac2-8c0e-cac80a534392)
    Message: Network map updated
    Time: 44 minutes, 45 seconds ago
  [INFO] SYSTEM (20d1991d-8aaa-440d-8c3d-e635708f11f8)
    Message: Network map updated
    Time: 17 minutes, 36 seconds ago
OS: linux/amd64
Daemon version: 0.60.5
CLI version: 0.60.5
Profile: default
Management: Connected to https://api.netbird.io:443
Signal: Connected to https://signal.netbird.io:443
Relays: 
  [stun:stun.netbird.io:443] is Available
  [stun:stun.netbird.io:5555] is Available
  [turns:turn.netbird.io:443?transport=tcp] is Available
  [rels://streamline-es-mad1-0.relay.netbird.io:443] is Available
Nameservers: 
FQDN: cluster.netbird.cloud
NetBird IP: 100.99.137.202/16
Interface type: Kernel
Quantum resistance: false
Lazy connection: false
SSH Server: Enabled (1 active session)
  [sesterce@100.99.150.233:47522] <interactive shell>
Networks: -
Forwarding rules: 0
Peers count: 5/8 Connected

Create and upload a debug bundle, and share the returned file key:

  netbird debug for 1m -AS -U

f79e391890ab27fb37c88b3b4be7011e22aa2e5ca6f38ffa9c4481884941f726/dd2c6c3c-7136-4e77-a914-f1a19b986ca9

*Uploaded files are automatically deleted after 30 days.*

**Screenshots**

**Additional context**

User.Lookup() function is able to resolve the remote user correctly on the server. I'm trying to understand what is doing the user check on the server side and then how it can be update to handle this kind of users.

**Have you tried these troubleshooting steps?**
- [x] Reviewed [client troubleshooting](https://docs.netbird.io/how-to/troubleshooting-client) (if applicable)
- [ ] Checked for newer NetBird versions
- [x] Searched for similar issues on GitHub (including closed ones)
- [x] Restarted the NetBird client
- [x] Disabled other VPN software
- [x] Checked firewall settings

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

When CGO is disabled (`CGO_ENABLED=0`), Go's `os/user` only reads `/etc/passwd` and `/etc/group` directly, bypassing the system's NSS stack. This means users provided by SSSD, LDAP, Active Directory, or other NSS modules can't be resolved.

This adds `getent passwd` / `id -G` fallback for user lookup, group resolution, and shell detection. These commands go through the host's NSS stack regardless of how the Go binary was built. The fallback is only compiled in non-CGO builds (or when `osusergo` tag is set), since libc handles NSS natively when CGO is enabled.

## Issue ticket number and link

Closes #4919

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [ ] Is a feature enhancement
- [ ] It is a refactor
- [x] Created tests that fail without the change (if possible)

> By submitting this pull request, you confirm that you have read and agree to the terms of the [Contributor License Agreement](https://github.com/netbirdio/netbird/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

## Documentation
Select exactly one:

- [ ] I added/updated documentation for this change
- [x] Documentation is **not needed** for this change (explain why)

No user-facing API or configuration changes. Behavior is transparent.

### Docs PR URL (required if "docs added" is checked)
Paste the PR link from https://github.com/netbirdio/docs here:

https://github.com/netbirdio/docs/pull/__

## Summary by CodeRabbit

* **New Features**
  * More reliable user/group lookups with platform- and build-aware fallbacks, improved input validation, 5s command timeouts, and explicit Windows behavior for shell lookup.
  * Enhanced shell detection with ordered fallbacks: passwd, NSS/getent, SHELL env, then default.

* **Tests**
  * Extensive cross-platform tests covering user lookup, group resolution, shell detection, parsing, validation, timeouts, and edge cases.

### CODE REVIEW

File: client/ssh/server/getent_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	}
+
+	current, err := user.Current()
+	require.NoError(t, err)
+
+	// getUserShell is the top-level function used by the SSH server.
+	shell := getUserShell(current.Uid)
+	require.NotEmpty(t, shell, "getUserShell must always return a shell")
+	assert.True(t, shell[0] == '/', "shell should be an absolute path, got %q", shell)
+}
```
[the remark this answers is not in the document]
Comment (lixmal) [reply client/ssh/server/getent_test.go:172]: extremely low risk this test will fail
